string part
string obj
---
bool ok
